lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/TsNode.cpp

```diff
@@ -134,6 +134,12 @@ std::string TsNode::ToString() const {
 torch::lazy::hash_t TsNode::GetOpHash(OpKind op,
                                       const lazy_tensors::Shape& shape,
                                       torch::lazy::hash_t hash_seed) {
+  if (lazy_tensors::Shape::IsDynamicMode()) {
+    torch::lazy::hash_t h = torch::lazy::HashCombine(
+        op.hash(), torch::lazy::Hash(shape.dim()));
```

Comment:
What's the reason to replace `shape.ToString()` with `shape.dim()` here for Dynamic Mode? This is perhaps related to the conversation in the parallel [PR](https://github.com/pytorch/pytorch/pull/72561) where I mentioned PT/XLA assumes `shape` represents tensor static upper bound numbers.

@Krovatkin